tests/test_fetch_contigs.py

```diff
@@ -0,0 +1,150 @@
+'''
+Copyright:
+
+University of Manitoba & National Microbiology Laboratory, Canada, 2020
+
+Written by: Arnab Saha Mandal
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+this work except in compliance with the License. You may obtain a copy of the
+License at:
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+'''
+
+# Importing functions from fetch_contigs.py 
+from n50.fetch_contigs import \
+    entrez_api, urls_array, \
+    download_needed, url_download
+
+from pathlib import Path
+import pytest
+import tempfile
+import urllib
+import os
+import sys
+
+START_DIR = Path(__file__).parent.absolute()
+
+# Defining parameters for valid NCBI api call
+test_genus_good = 'Escherichia'
+test_species_good = 'coli'
+test_num_good = 3
+test_email = 'idontexist@gmail.com'
+test_api_key = None
+
+# Defining return objects for sucessful test functions
+TEST_ID_LIST = ['7143261', '7143111', '7140871']
+TEST_URL_LIST = [
+    'ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/013/112/445/GCA_013112445.1_ASM1311244v1',
+    'ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/013/106/615/GCA_013106615.1_PDT000734164.1',
+    'ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/013/106/435/GCA_013106435.1_PDT000733736.1',
+]
+
+TEST_FAKE_URL = 'ftp://fantastic_organism_genomic.fna.gz'
+
+
+# Defining mockheader for fake url request
+class MockHeader:
+    headers = {"Content-Length": 0}
+
+    def read(self):
+        return ""
+
+
+class TestFetchContigs:
+    @classmethod
+    def setup_class(cls):
+        sys.path.append(os.path.dirname(os.path.abspath(__file__)))
+
+
+    # Test for good API call
+    def test_entrez_api_good(self):
+        method_id_list = entrez_api(test_genus_good, test_species_good,
```

Comment:
you should be mocking the `Entrez.esearch` function so that you don't rely on the external api for your tests to pass.

Reply:
See this as an example https://github.com/phac-nml/irida-galaxy-importer/blob/development/irida_import/tests/unit/test_irida_import.py#L383